### DISCUSSION TITLE
Scatter extension question

### DISCUSSION THREAD

**Discussion by cameronosmith** [pyg-team/pytorch_geometric#2509] [Q&A] [answered]
Hi, hope you're doing well.

I have a set of clusters of irregular size, and would like to apply a different function to each of these clusters. More specifically, the clusters are sets of points in different coordinate frames and I need to apply a transformation matrix to the points. Is there a scatter-like function or technique where I can assign each of the clusters a different transformation matrix to apply or am I doomed to either iterate each cluster manually or tile the transformation matrix across the set of all clusters points?

**Comment by rusty1s** [comment, marked as answer]
I guess so :( I don't think there is an elegant way to parallelize that in a SIMD-based architecture like CUDA. Perhabs you can get some speed ups via Dynamic Parallelism in TorchScript: https://pytorch.org/tutorials/advanced/torch-script-parallelism.html.

**Reply by cameronosmith** [reply]
Got it. Thank you for your response and the PyTorch parallelization reference. Take care.
